anomalib/training/strategies/default/validate.py

```diff
@@ -0,0 +1,77 @@
+"""Validation loop."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+from functools import lru_cache
+from typing import Any, List
+
+from pytorch_lightning.loops.dataloader.evaluation_loop import EvaluationLoop
+from pytorch_lightning.loops.epoch.evaluation_epoch_loop import EvaluationEpochLoop
+from pytorch_lightning.utilities.types import EPOCH_OUTPUT, STEP_OUTPUT
+
+import anomalib.training as core
+
+
+class AnomalibValidationEpochLoop(EvaluationEpochLoop):
+    def __init__(self):
+        super().__init__()
+        self.trainer: core.AnomalibTrainer
+
+    def _evaluation_step_end(self, *args, **kwargs) -> STEP_OUTPUT | None:
+        """Runs ``validation_step_end`` after the end of one validation step."""
+        outputs = super()._evaluation_step_end(*args, **kwargs)
+        if outputs is not None:
+            self.trainer.post_processor.outputs_to_cpu(outputs)
+            self.trainer.post_processor.post_process(outputs)
+        return outputs
+
+    def _on_evaluation_batch_end(self, output: STEP_OUTPUT, **kwargs: Any) -> None:
+        """The ``on_validation_batch_end`` hook.
+
+        Args:
+            output: The output of the performed step
+            batch: The input batch for the step
+            batch_idx: The index of the current batch
+            dataloader_idx: Index of the dataloader producing the current batch
+        """
+        super()._on_evaluation_batch_end(output, **kwargs)
+        self.trainer.normalizer.update_metrics(output)
```

Comment:
What is the difference between `self.trainer.post_processor.update_metrics` vs `self.trainer.normalizer.update_metrics`?

Reply:
Maybe I should make it more explicit. The `post_processor` updates the threshold metrics whereas the `normalizer` updates the min-max/cdf metrics. How about I rename it to `post_processor.update_threshold_metrics`?